dmoj/problem.py

```diff
@@ -60,6 +60,10 @@ def __init__(self, problem_id, time_limit, memory_limit, meta):
         self.problem_data.archive = self._resolve_archive_files()
         self._resolve_test_cases()
 
+    @classmethod
+    def get_config(cls, problem_id):
+        return Problem(problem_id, None, None, {}).config
```

Comment:
This is a bit gross... I think what we want is a `ProblemConfig` whose `__init__` takes a `data : ProblemDataManager` and `meta : ConfigNode = None`. Then this `get_config` doesn't need to exist, and `Problem.__init__` can just construct the `ProblemConfig`. The `ProblemConfig` can probably just be a subclass of `ConfigNode`.

Reply:
That would effectively kick this grossness to the command (the `ProblemDataManager` requires the `Problem` in its constructor)

Reply:
All `ProblemDataManger` needs is the `root_dir` from the `Problem`, which you have already.